types/index.d.ts

```diff
@@ -0,0 +1,45 @@
+export interface Configuration {
+  options?: Options;
+  templates?: Template[];
+  vocabulary?: Vocabulary;
+}
+
+export interface DefinitelyOptions {
```

Comment:
The naming here seems maybe a bit weird 😛 
Maybe something like `ResolvedOptions` instead?

Reply:
Yeah, it's a bit weird, but between `DefinitelyOptions` and something like `ResolvedOptions` I figured I'd go for "definitely" as a silly reference to "DefinitelyTyped", which is a ts thing. :P 

`ResolvedOptions` does sound a lot better though.